datafusion/catalog/src/dynamic_file/catalog.rs

```diff
@@ -0,0 +1,184 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! dynamic_file contains [`DynamicFileCatalog`] that creates tables from file paths
+//! if the wrapped [`CatalogProviderList`] doesn't have the table provider.
```

Comment:
I suggest making module level documentation (which is rendered typically as a single line) point to the main struct/trati

```suggestion
//! [`DynamicFileCatalog`] that creates tables from file paths
```